libraries/Resample/design_fir.py

```diff
@@ -0,0 +1,76 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.signal as sig
+import os
+
+sr = 1                                 # samplerate
+ripple = {"high": 80, "low": 60}       # stopband attenuation and passband ripple
+cutoff_ny = {"high": 0.9, "low": 0.7}  # cutoff relative to nyquist
+
+fname = "aa_fir.h"
+try:
+    os.remove(fname)
+except FileNotFoundError:
+    pass
+file = open(fname, "a")
+
+file.write("#include <vector>\n\n")
+
+for down in [2, 4, 8, 16]:
+    for quality in ripple:
+        nyquist_resampled = sr/down/2                 # new nyquist
+        cutoff = 0.5 * sr/down * cutoff_ny[quality]   # filter cutoff
+        width = (nyquist_resampled - cutoff)*2     # filter transition bandwidth
+
+        # design kaiser such that stopband starts at nyquist
+        numtaps, beta = sig.kaiserord(ripple[quality], width/(0.5*sr))
+        taps = sig.firwin(numtaps, cutoff, window=('kaiser', beta), fs=sr)
+        tapsmin = sig.minimum_phase(taps)
+
+        taps = taps/np.sum(taps)
+        tapsmin = tapsmin/np.sum(tapsmin)
+
+        w, h = sig.freqz(taps, worN=8000)
+        w, hmin = sig.freqz(taps, worN=8000)
+
+        f = w * 0.5*sr/np.pi  # Convert w to Hz.
+
+        plt.figure(1)
```

Comment:
do these `plt.xxx` commands work fine on Bela? I am wondering if it'll crash because of lack of display. In the latter case they should probably made conditional.

Reply:
The script was meant to be run on a system that has `scipy` installed, see above comment. Is that a problem?